### PR TITLE
feat: update to undici v7

### PR DESCRIPTION
**Pull request by ivan-tymoshenko**

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Review by mcollina — APPROVED** [review]
lgtm

File: index.js [hunk 1]
```diff
@@ -92,36 +99,30 @@ function createThreadInterceptor (opts) {
...
-          } else {
-            headers.push(key)
-            headers.push(value)
+        try {
+          handler.onRequestStart(controller, {})
+          if (controller.aborted) {
+            handler.onResponseError(controller, controller.reason)
+            return
           }
+          handler.onResponseStart(
```
Comment (ivan-tymoshenko) [inline index.js:113]: Can the response be aborthed in the `onResponseStart` (I think so)? Do we need a `controller.aborted` chech after this call?

Comment (mcollina) [reply index.js:113]: I've added the lines and skipped the tests.

**Pull request merged**